lib/logpipe.h

```diff
@@ -363,6 +366,16 @@ log_pipe_append(LogPipe *s, LogPipe *next)
   s->pipe_next = next;
 }
 
+void
+log_pipe_set_persist_id(LogPipe * const self, const gchar * const persist_id);
+
+static inline void
+log_pipe_generate_persist_id(LogPipe * const self)
+{
+  if (self->persist_id != NULL && self->generate_persist_id != NULL)
+    self->persist_id = self->generate_persist_id(self);
```

Comment:
The type of the return value and the type of the field are the same (`const gchar *` and `gchar const *`), however, mixing the const related conventions is rarely a good idea.

Fixed in cc3bf16146493ed2e8ceb516d0fdab08763265f3.